internal/app/upstream/client.go

```diff
@@ -0,0 +1,57 @@
+package client
```

Comment:
Thoughts on naming this package `upstream` and the interface `Client`? So it's `upstream.Client` rather than `client.XdsClient`?

Reply:
renamed the package to upstream